### PR TITLE
128

### PR DESCRIPTION
**Pull request by mtmcduffie**

### CODE REVIEW

**Comment by mtmcduffie** [comment]
**/oauth/debug/token** returns

```
{
  "access_token": "XZL6YCYJV73DRU8K4A5I", 
  "refresh_token": "24U3NNBKK8KTFQ45IMZM"
}
```

**/oauth/debug/introspect** returns
```
{
  "access_token": "XZL6YCYJV73DRU8K4A5I", 
  "approval_expires": "Thu, 01 Jan 1970 00:06:00 GMT", 
  "expires": "Thu, 29 Mar 2018 18:20:01 GMT", 
  "refresh_token": "24U3NNBKK8KTFQ45IMZM", 
  "scopes": [
    "patient", 
    "allergies"
  ], 
  "security_labels": [
    "patient", 
    "allergies"
  ], 
  "token_type": "Bearer"
}
```

File: auth_proxy/services/oauth.py [hunk 1]
```diff
@@ -199,3 +201,37 @@ def cb_tokensetter(self, token, request, *args, **kwargs):
...
+            user=creating_user,
+            patient_id=patient_id
+        )
+
+        self.db.session.add(token)
+        self.db.session.commit()
+
+        character_options = string.ascii_uppercase + string.digits
+
+        generated_access_token = ''.join(random.SystemRandom().choice(character_options) for _ in range(20))
```
[the remark this answers is not in the document]
Comment (mtmcduffie) [reply auth_proxy/services/oauth.py]: Thank you, Much cleaner looking.

File: auth_proxy/services/oauth.py [hunk 2]
```diff
@@ -199,3 +201,35 @@ def cb_tokensetter(self, token, request, *args, **kwargs):
...
+            patient_id=patient_id
+        )
+
+        self.db.session.add(token)
+        self.db.session.commit()
+
+        generated_access_token = str(uuid.uuid4())
+        generated_refresh_token = str(uuid.uuid4())
+
+        new = token.refresh(generated_access_token,
```
Comment (jmandel) [inline auth_proxy/services/oauth.py]: Can you add a quick comment documenting why refresh step is needed?

Comment (mtmcduffie) [reply auth_proxy/services/oauth.py]: Done! It's mostly for future proofing if we implement logic in the refresh that we'd want to be applied to 'debug' tokens too.

File: auth_proxy/services/oauth.py [hunk 3]
```diff
@@ -5,11 +5,12 @@
 
 import arrow
 import flask_login
+import random
```
Comment (erikwiffin) [inline auth_proxy/services/oauth.py]: You don't need these anymore.

**Pull request merged**